source/retrieve-data.txt

```diff
@@ -0,0 +1,196 @@
+filter that specifies the criteria of the document you want to find. Then, chain
+the ``first()`` method to find a single document. If more than one document matches the query
+filter, this construct returns the *first* matching document from the retrieved
+results. If no documents match the query filter, the method returns
+``None``.
+
+.. tip::
+   
+   The ``find().first()`` construct is useful when you know there's only one matching document,
+   or you're only interested in the first match.  
+
+The following example uses the ``find().first()`` construct to find the first document where
+the ``"cuisine"`` field has the value ``"Bakery"``:
+
+.. code-block:: java
+   :copyable: true
+
+   Publisher<Document> findDocPublisher = restaurants.find(
+       eq("cuisine", "Bakery")).first();
+   List<Document> findResult = Flux.from(findDocPublisher).collectList().block();
+   
+.. tip:: Sort Order
+
+   The ``find().first()`` construct returns the first document in 
+   :manual:`natural order </reference/glossary/#std-term-natural-order>`
+   on disk if no sort criteria is specified.
+
+.. TODO: uncomment when sort guide made
+.. To learn more about sorting, see the :ref:`sort guide <java-rs-sort>`.
+
+.. _java-rs-retrieve-find-multiple:
+
+Find Multiple Documents
+~~~~~~~~~~~~~~~~~~~~~~~
+
+To find multiple documents in a collection, pass a query filter to the ``find()``
+method that specifies the criteria of the documents you want to retrieve.
+
+The following example uses the ``find()`` method to find all documents where
+the ``"cuisine"`` field has the value ``"Spanish"``:
+
+.. code-block:: java
+   :copyable: true
+
+   FindPublisher<Document> findDocPublisher = restaurants.find(
+           eq("cuisine", "Spanish"));
+   List<Document> findResult = Flux.from(findDocPublisher).collectList().block();
+
+You can iterate over the documents you retrieved using the ``find()`` method by using a ``for`` loop, as shown in
+the following example:
+
+.. code-block:: java
+   :copyable: true
+
+   FindPublisher<Document> findDocPublisher = restaurants.find(
+           eq("cuisine", "Spanish"));
+   List<Document> findResult = Flux.from(findDocPublisher).collectList().block();
+   
+   for (Document document : findResult) {
+   ...
+   }
+
+.. note:: Find All Documents
+
+   To find all documents in a collection, pass no parameters to the ``find()`` method:
+
+   .. code-block:: java
+
+      Publisher<Document> findAllPublisher = restaurants.find();
+      List<Document> findResult =
+              Flux.from(findAllPublisher).collectList().block();
+      
+
+Modify Find Behavior
+~~~~~~~~~~~~~~~~~~~~
+
+You can modify the behavior of the ``find()`` method by chaining methods
+to it. The following table describes commonly used methods:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Argument
+     - Description
+
+   * - ``batchSize(int batchSize)`` 
+     - | Limits the number of documents to hold in a cursor at a given time.
```

Comment:
Q: At this point in the docs, do users have the context to understand cursors? If not, maybe link to the Server docs or the Java RS page where this concept is explained.